src/prover/mod.rs

```diff
@@ -281,43 +295,112 @@ impl Prover {
                 unreachable!();
             }
-            CircuitType::Chunk => {
+            ProofType::Chunk => {
                 let chunk_task_detail: ChunkTaskDetail = serde_json::from_str(&task.task_data)?;
                 let serialized_traces = self
                     .l2geth_client
                     .as_ref()
                     .unwrap()
-                    .get_sorted_traces_by_hashes(&chunk_task_detail.block_hashes)
+                    .get_traces_by_hashes(&chunk_task_detail.block_hashes)
                     .await?;
                 // Note: Manually join pre-serialized traces since they are already in JSON format.
                 // Using serde_json::to_string would escape the JSON strings, creating invalid nested JSON.
                 let input = format!("[{}]", serialized_traces.join(","));
 
                 Ok(ProveRequest {
-                    circuit_type: task.task_type,
+                    proof_type: task.task_type,
                     circuit_version: self.circuit_version.clone(),
                     hard_fork_name: task.hard_fork_name.clone(),
                     input,
                 })
             }
-            CircuitType::Batch | CircuitType::Bundle => Ok(ProveRequest {
-                circuit_type: task.task_type,
+            ProofType::Batch | ProofType::Bundle => Ok(ProveRequest {
+                proof_type: task.task_type,
+                circuit_version: self.circuit_version.clone(),
+                hard_fork_name: task.hard_fork_name.clone(),
+                input: task.task_data.clone(),
+            }),
+        }
+    }
+
+    #[cfg(feature = "openvm")]
+    async fn build_openvm_input(&self, task: &GetTaskResponseData) -> anyhow::Result<ProveRequest> {
+        match task.task_type {
+            ProofType::Chunk => {
+                let chunk_task_detail: ChunkTaskDetail = serde_json::from_str(&task.task_data)?;
+                let mut witnesses = vec![];
+                for block_hash in chunk_task_detail.block_hashes {
+                    witnesses.push(self.build_block_witness(block_hash).await?);
+                }
+                witnesses.sort_by(|a, b| a.header.number.cmp(&b.header.number));
+
+                Ok(ProveRequest {
+                    proof_type: task.task_type,
+                    circuit_version: self.circuit_version.clone(),
+                    hard_fork_name: task.hard_fork_name.clone(),
+                    input: serde_json::to_string(&witnesses)?,
+                })
+            }
+            ProofType::Batch | ProofType::Bundle => Ok(ProveRequest {
+                proof_type: task.task_type,
                 circuit_version: self.circuit_version.clone(),
                 hard_fork_name: task.hard_fork_name.clone(),
                 input: task.task_data.clone(),
             }),
+            _ => bail!("unsupported task type: {:?}", task.task_type),
         }
     }
+
+    #[cfg(feature = "openvm")]
+    async fn build_block_witness(
```

Comment:
The openvm prover required a `BlockWitness` data struct which now can be [constructed from the utility in sbv](https://github.com/scroll-tech/stateless-block-verifier/blob/316e6c77ed3477f5b800e5f794eace412de717e2/crates/utils/src/commands/witness/dump.rs#L71), not only the block data. So we still need quite a few works to be added here.

Reply:
What do you mean? I use DumpWitnessCommand already.